var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -60,6 +61,19 @@ def install(self, spec, prefix):
         if spec.satisfies("@4.7.1:") and sys.platform != 'darwin':
             enabled_languages.add('go')
 
+        # Fix a standard header file for OS X Yosemite & earlier that
+        # is GCC incompatible by replacing non-GCC compliant macros
+        if sys.platform == 'darwin':
```

Comment:
you write `Yosemite` and earlier, why don't you check for that (looking at versions)? I am not sure this patch will work on elcapitan or sierra. Generally what Apple ships with next versions is unknown. So i would adjust the `if` so that it is in accordance with the comment.